### PR TITLE
Display phenotypes

### PR DESCRIPTION
**Pull request by moonso**
This PR adds a functionality or fixes a bug.

**How to test**:
1. Install this branch
1. Choose a variant and see that the omim phenotypes are visible

**Expected outcome**:
All OMIM phenotypes should now be visible

**Review:**
- [x] code approved by @dnil 
- [x] tests executed by moonso

### CODE REVIEW

**Comment by moonso** [comment]
Ok it would be great to get this one merged, could you please have a look. It might seem like there are a lot of changes but most of the are code that's being splitted into files.

**Comment by moonso** [comment]
Kind of a nightmare when files have been blacked in different ways... how can we avoid this?

**Comment by northwestwitch** [comment]
> Kind of a nightmare when files have been blacked in different ways... how can we avoid this?

Oh no which black version do you have?

**Comment by moonso** [comment]
> Oh no which black version do you have?
`black, version 19.10b0`

**Comment by northwestwitch** [comment]
> > Oh no which black version do you have?
> 
> `black, version 19.10b0`

It should be fine, don't worry! I see that this PR is mostly HTML code anyway, so it shouldn't be a problem. I can review tomorrow probably, but I see that you still have some problem with the tests!

**Comment by dnil** [comment]
We wrote some of it together, so at least the commits from Monday are well vetted. I will be unresponsive with clinic meetings tomorrow, but could have a bit of time in the afternoon.

**Comment by moonso** [comment]
Tests run perfect on my local. No idea why the command test fails on travis...

**Comment by northwestwitch** [comment]
I'll test it later this morning, I'm not home at the moment

File: scout/server/blueprints/variant/templates/variant/utils.html [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+      {% endfor %}
+    </div>
+    </div>
+  </div>
+{% endmacro %}
+
+{% macro rankscore_panel(variant) %}
+  <div class="card panel-default">
+    <div class="panel-heading">Rank score results</div>
+    <div class="table-responsive">
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/utils.html]: card body?

File: scout/server/blueprints/variant/templates/variant/utils.html [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+{% endmacro %}
+
+{% macro overlapping_panel(variant, overlapping_vars, case, institute) %}
+  <div class="card panel-default">
+    {% if variant.category == 'sv' %}
+      <div class="panel-heading">Overlapping SNVs</div>
+    {% else %}
+      <div class="panel-heading">Overlapping SVs</div>
+    {% endif %}
+    <div class="table-responsive">
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/utils.html]: body? The margins are I suppose a matter of taste; perhaps we could have margin-less class for the card body as well, if we want the cards to be complete, but no margins?

File: scout/server/blueprints/variant/templates/variant/utils.html [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+    </div>
+    <div class="card-body">
+    <div class="table-responsive">
+      <table class="table table-bordered">
+        <thead>
+          <tr>
+            <th rowspan="2">Sample</th>
+            <th rowspan="2">Genotype (GT)</th>
+            <th colspan="2" title="Unfiltered count of reads that support a given allele.">Allele depth (AD)</th>
+            <th rowspan="2" colspan="1" title="Phred-scaled confidence that the true genotype is the one provided in GT (max=99).">Genotype quality (GQ)</th>
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/utils.html]: This is not so elegant - or true - right now with SV "GQ"? But, if that other issue is solved quickly, I suppose we could live with it.
Screenshot 2020-02-05 at 16 36 31

File: scout/server/blueprints/variant/templates/variant/sv-variant.html [hunk 4]
```diff
@@ -37,307 +38,174 @@
...
+                  </form>
+                </div>
+              </div>
+              <div class="list-group mt-3">
+                <div>
+                  <form action="{{ url_for('variant.variant_update', institute_id=institute._id, case_name=case.display_name, variant_id=variant._id) }}" method="POST">
+                    <label class="ml-3">Dismiss variant</label>
+                    <div class="row">
+                      <div class="col-8 d-flex">
+                          <select multiple class="ml-3 selectpicker" title="Select categories..." name="dismiss_variant" id="dismiss_variant" >
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/sv-variant.html]: Right, this popped back again:
Screenshot 2020-02-05 at 16 39 37

File: scout/server/blueprints/variant/templates/variant/variant.html [hunk 5]
```diff
@@ -70,17 +71,21 @@
...
         <div class="col-sm">{{ inheritance_panel() }}</div>
-        <div class="col-sm">{{ omim_phenotypes() }}</div>
+        <div class="col-sm">{{ omim_phenotypes(variant) }}</div>
     </div>
     <div class="row">
-      <div class="col-sm">{{ frequency_list() }}</div>
-      <div class="col-sm">{{ severity_list() }}</div>
+      <div class="col">{{ frequencies(variant) }}</div>
+      {% if config['LOQUSDB_SETTINGS'] %}
+        <div class="col">{{ observations_panel(variant, observations) }}</div>
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant.html]: observations_panel needs case.

File: scout/server/blueprints/variant/templates/variant/utils.html [hunk 6]
```diff
@@ -0,0 +1,540 @@
...
+            {% endif %}
+          </div>
+        </div>
+      </li>
+    </ul>
+  </div>
+  </div>
+{% endmacro %}
+
+{% macro observations_panel(variant, observations) %}
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/utils.html]: ...and case here as well.

File: scout/server/blueprints/variant/templates/variant/variant.html [hunk 7]
```diff
@@ -70,17 +71,21 @@
...
+      {% endif %}
+      <div class="w-100"></div>
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant.html]: 👀

File: scout/server/blueprints/cases/templates/cases/case.html [hunk 8]
```diff
@@ -1,10 +1,14 @@
 {% extends "layout_bs4.html" %}
 {% from "cases/collapsible_actionbar.html" import action_bar, research_modal, rerun_modal %}
-{% from "utils.html" import comments_panel, activity_panel %}
-{% from "cases/utils.html" import individuals_table %}
+{% from "utils.html" import comments_panel, activity_panel, pedigree_panel %}
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/blueprints/cases/templates/cases/case.html]: Yes I started to do that for case

File: scout/server/blueprints/variant/templates/variant/variant.html [hunk 9]
```diff
@@ -886,8 +727,10 @@ <h6>No matching causative variants</h6>
         </tbody>
       </table>
       {% endif %}
+  </div>
 
-      <h6 class="card-header">Gene models:</h6>
+  <div class="card mt-3 panel-default">
+    <div class="panel-heading">Gene models:</div>
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant.html]: No colon!

File: scout/server/config.py [hunk 10]
```diff
@@ -17,11 +17,10 @@
 MAIL_USE_SSL = False
 
 # connection details for LoqusDB MongoDB database
-# LOQUSDB_SETTINGS = dict(
-#     binary='/path/to/bin/loqusdb',
-#     config_path=<path/to/loqus/config>
-#
-# )
+LOQUSDB_SETTINGS = dict(
+    binary_path="/Users/mansmagnusson/miniconda3/envs/loqusdb/bin/loqusdb",
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/config.py]: Hehe

File: scout/server/blueprints/cases/controllers.py [hunk 11]
```diff
@@ -244,7 +244,7 @@ def case(store, institute_obj, case_obj):
 
     data = {
         "status_class": STATUS_MAP.get(case_obj["status"]),
-        "other_causatives": store.check_causatives(case_obj=case_obj),
+        "other_causatives": [var for var in store.check_causatives(case_obj=case_obj)],
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/blueprints/cases/controllers.py]: The function just returns a cursor

File: scout/server/blueprints/variant/utils.py [hunk 12]
```diff
@@ -8,12 +8,12 @@
 
 def add_panel_specific_gene_info(panel_info):
     """Adds manualy curated information from a gene panel to a gene
-    
+
```
Comment (dnil) [inline scout/server/blueprints/variant/utils.py]: Is this black versions, or editor malfunctions?

File: scout/server/blueprints/variant/templates/variant/utils.html (hunk 3 above)
Comment (moonso) [reply scout/server/blueprints/variant/templates/variant/utils.html]: Mm and a bit out of scope for this PR

File: scout/server/blueprints/variant/templates/variant/variant.html (hunk 5 above)
Comment (moonso) [reply scout/server/blueprints/variant/templates/variant/variant.html]: Thx

**Comment by moonso** [comment]
@dnil I think it is the new loqus functionality that makes the variant page for SVs slow. I need to spend some time to investigate it. But it is not this PR

**Comment by dnil** [comment]
> @dnil I think it is the new loqus functionality that makes the variant page for SVs slow. I need to spend some time to investigate it. But it is not this PR

Yiekes! Ok, thanks!

File: scout/server/blueprints/variant/templates/variant/variant.html (hunk 7 above)
Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant.html]: Causes this I suppose?
Screenshot 2020-02-05 at 16 55 24

File: scout/server/blueprints/variant/templates/variant/utils.html [hunk 13]
```diff
@@ -0,0 +1,540 @@
...
+  <div class="card panel-default mt-3">
+    <div class="panel-heading">
+      <a href="https://github.com/moonso/loqusdb" target="_blank">Local observations</a>
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/blueprints/variant/templates/variant/utils.html]: Fixed

File: scout/server/blueprints/variant/templates/variant/variant.html (hunk 7 above)
Comment (moonso) [reply scout/server/blueprints/variant/templates/variant/variant.html]: :)

File: scout/server/blueprints/variant/utils.py (hunk 12 above)
Comment (moonso) [reply scout/server/blueprints/variant/utils.py]: BLAAAAACK

File: scout/server/blueprints/variant/templates/variant/variant.html (hunk 9 above)
Comment (moonso) [reply scout/server/blueprints/variant/templates/variant/variant.html]: Fixed!

File: scout/server/blueprints/variant/templates/variant/gene_disease_relations.html [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+            <tr>
+              <th>Gene</th>
+              <th>Phenotype</th>
+              <th>Inheritance model</th>
+            </tr>
+          </thead>
+          <tbody>
+            {% for gene in variant.genes %}
+              {% if gene.common %}
+                {% for disease_term in gene.common.phenotypes %}
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/gene_disease_relations.html]: :+1:

File: scout/server/blueprints/variant/templates/variant/sv-variant.html [hunk 15]
```diff
@@ -202,7 +202,7 @@
   <div class="row">
     <div class="col-md-4">
     {{ frequencies(variant) }}
-    </div>
+  </div>
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/sv-variant.html]: HTML indentation error!!

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/sv-variant.html]: :wink:

**Comment by dnil** [comment]
dev dep medocs: https://github.com/squidfunk/mkdocs-material/issues/1331

**Comment by moonso** [comment]
Ok stuff is fixed! I'm merging this now

**Pull request merged**